### ISSUE TITLE
incorrect adjustment of error term when using lscov

### ISSUE THREAD

**Issue by tommysprague** [vistalab/vistasoft#220]
When computing retinotopic RF models (specifically, I've worked with rmGridFit_oneGaussianNonlinear.m, though this problem seems general across many), the grid fit step uses lscov to simultaneously estimate betas for the model predictor and trends, as well as a measurement of model error. lscov, as addressed in the comments, returns the mean squared error, rather than sum squared error, which is used throughout vistasoft, especially when computing variance explained. 

This is meant to be corrected after looping through model iterations, using the snippet (lines 97-101):
```
% Correct lscov. It returns the mean rss. To maintain compatibility with the
% sum rss this function expects, we have to multiply by the divisor. See
% the lscov docs for details.
model.rss=single(model.rss.*(size(prediction,1)-size(trends,2)+1)); 
```

In a model with 3 stimulus types (bar sizes), 120 TRs per stimulus type, our "prediction" is 360 TRs long, and trends is 360 x 3 (constant offset for each stimulus type). This works out to correcting based on: n_TRs (360) - n_trends (3) + 1 = 358, in this case.

But in checking how lscov itself computes MSE from the raw RSS, it seems as though it corrects based on the rank of the predictor matrix. In the call to lscov (line 58), rmGridFit_*.m functions pass the current model prediction horizontally concatenated with the trends. In my example, this results in a 360 x 4 matrix.

The relevant portion of lscov.m for a full-rank model is copied below (lines 217-228)
```% Compute the MSE, need it for the std. errs. and covs.
        dfe = nobs - rankA;
        if dfe > 0
            if issparse(A)
                mse = full(sum(b.*conj(b),1) - sum(z.*conj(z),1)) ./ dfe;
            else
                if rankA < nvar
                    Q = Q(:, keepCols);
                end
                res = b - Q*z;
                mse = full(sum(res.*conj(res),1)) ./ dfe;  % TCS::: correction applied here!!!!!
            end
```

Because the predictor matrix is full rank (4), the corrected MSE term is scaled by the # of observations (360) - rank (4) = 356. 

This means that the correction applied to the lscov-derived error is slightly inflated; instead of multiplying error by 356, in this case, it's multiplying it by 358. This will result in lower VE, and, potentially in some cases, negative VE. 

I believe the correct scaling factor is:
`model.rss=single( model.rss.* ( size(prediction,1) - (size(trends,2)+1) ) );` 

I haven't checked every single rmGridFit_*.m function, but a spot check of a few which use lscov.m (e.g., rmGridFit_oneGaussian.m) reveals that they use the similar incorrect correction factor.

**Comment by tommysprague** [comment]
(note: this is with MATLAB R2016b, in case there are version-specific differences in lscov behavior)

**Comment by JWinawer** [comment]
OK, fixed.

**Comment by lmperry** [comment]
Closed via 348d2818c4eb372db10a1c3189dec37114e8f8c8
